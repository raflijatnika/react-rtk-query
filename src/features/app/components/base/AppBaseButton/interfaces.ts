// Antd
import { ButtonProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseButtonProps extends ButtonProps {
  backgroundColor?: APP_COLOR | APP_COLOR_LIGHT
  fontColor?: APP_COLOR | APP_COLOR_LIGHT
  width?: number
  height?: number
}
