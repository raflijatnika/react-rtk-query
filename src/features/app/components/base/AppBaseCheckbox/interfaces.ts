// Antd
import { CheckboxProps } from 'antd'

export type IAppBaseCheckboxProps = CheckboxProps
