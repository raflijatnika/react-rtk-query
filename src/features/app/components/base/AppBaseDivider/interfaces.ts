// Antd
import { DividerProps } from 'antd'

export type IAppBaseDividerProps = DividerProps
