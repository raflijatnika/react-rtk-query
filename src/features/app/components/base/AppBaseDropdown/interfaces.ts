// React
import { ReactElement } from 'react'

// Interfaces
import { DropdownProps, MenuProps } from 'antd'

export interface IDropdownProps {
  children?: ReactElement
  trigger?: DropdownProps['trigger']
  items: MenuProps['items']
}
