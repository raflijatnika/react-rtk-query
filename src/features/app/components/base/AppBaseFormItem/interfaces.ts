// Interfaces
import { FormItemProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseFormItemProps extends FormItemProps {
  labelColor?: APP_COLOR | APP_COLOR_LIGHT
  labelSize?: number
}
