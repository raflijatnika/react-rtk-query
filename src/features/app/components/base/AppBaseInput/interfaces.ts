// Interfaces
import { InputProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseInputProps extends InputProps {
  backgroundColor?: APP_COLOR | APP_COLOR_LIGHT
  height?: number
  width?: number
}
