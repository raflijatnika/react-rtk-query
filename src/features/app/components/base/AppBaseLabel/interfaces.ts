// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseLabelProps {
  fontWeight?: number
  fontSize?: number
  isBold?: boolean
  fontColor?: APP_COLOR | APP_COLOR_LIGHT
}
