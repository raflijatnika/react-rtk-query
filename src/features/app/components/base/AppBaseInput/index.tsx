// Styled Components
import styled from 'styled-components'

// Antd
import { Input } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

// Interfaces
import { IAppBaseInputProps } from '@/features/app/components/base/AppBaseInput/interfaces'

const AppBaseInput = styled(Input)<IAppBaseInputProps>`
  /* &:not(.ant-input-status-error) {
    border: none;
  }

  background-color: ${props =>
    props?.backgroundColor || APP_COLOR_LIGHT.BACKGROUND};
  height: ${props => (props?.height ? `${props.height}px` : '50px')};
  width: ${props => (props?.width ? `${props.width}px` : '100%')};
  border-radius: 10px;
  font-size: 14.22px;
  color: ${APP_COLOR.DARK}; */
`

const AppBaseInputSearch = styled(Input.Search)<IAppBaseInputProps>``

export { AppBaseInput, AppBaseInputSearch }
