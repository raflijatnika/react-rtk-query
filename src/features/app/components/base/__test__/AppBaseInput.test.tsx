// Test
import { render } from '@testing-library/react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import 'jest-styled-components'

// Components
import { AppBaseInput } from '../AppBaseInput'

// Interfaces
import { IAppBaseInputProps } from '@/features/app/components/base/AppBaseInput/interfaces'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

const rendererInstance = (props?: IAppBaseInputProps): ReactTestRenderer =>
  renderer.create(<AppBaseInput {...props} />)

const inputStyleOptions: jest.Options = {
  modifier: '&:not(.ant-input-status-error)'
}

it('AppBaseInput should render', () => {
  render(<AppBaseInput />)
})

it('AppBaseInput should have snapshot', () => {
  const tree = rendererInstance().toJSON()

  expect(tree).toMatchSnapshot()
})

it('AppBaseInput should have default style', () => {
  const tree = rendererInstance().toJSON()

  expect(tree).toHaveStyleRule('border', 'none', inputStyleOptions)
  expect(tree).toHaveStyleRule('background-color', APP_COLOR_LIGHT.BACKGROUND)
  expect(tree).toHaveStyleRule('height', '50px')
  expect(tree).toHaveStyleRule('width', '100%')
  expect(tree).toHaveStyleRule('border-radius', '10px')
  expect(tree).toHaveStyleRule('font-size', '14.22px')
  expect(tree).toHaveStyleRule('color', APP_COLOR.DARK)
})

it('AppBaseInput should render correct props', () => {
  const tree = rendererInstance({
    backgroundColor: APP_COLOR.DARK,
    width: 200,
    height: 100
  }).toJSON()

  expect(tree).toHaveStyleRule('background-color', APP_COLOR.DARK)
  expect(tree).toHaveStyleRule('width', '200px')
  expect(tree).toHaveStyleRule('height', '100px')
})
