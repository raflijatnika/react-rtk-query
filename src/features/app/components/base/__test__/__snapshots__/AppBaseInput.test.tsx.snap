// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBaseInput should have snapshot 1`] = `
.c0 {
  background-color: #F1F4FA;
  height: 50px;
  width: 100%;
  border-radius: 10px;
  font-size: 14.22px;
  color: #06152B;
}

.c0:not(.ant-input-status-error) {
  border: none;
}

<input
  className="ant-input c0"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  type="text"
  value=""
/>
`;
